.github/workflows/update_iree_requirement_pins.yml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2025 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: Update IREE requirement pins
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Weekdays at 11:00 AM UTC = 03:00 AM PST / 04:00 AM PDT
+    - cron: "0 11 * * 1-5"
+
+jobs:
+  update-iree:
+    if: ${{ github.repository_owner == 'nod-ai' || github.event_name != 'schedule' }}
+    runs-on: ubuntu-24.04
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: "Setting up Python"
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: 3.11
+
+      # This sets a few environment variables used below.
+      - name: Update IREE requirement pins
+        run: build_tools/update_iree_requirement_pins.py
+
+      - name: Create or update pull request
```

Comment:
did you have to set up keys / perms or anything like that to auto-create PRs in your own personal repo? or is it as simple as just setting the default branch to this?

Reply:
I have these settings in my forks:
![image](https://github.com/user-attachments/assets/881e7b46-b186-4534-8051-091b83d90038)

I see that we don't have the same settings in this repository though. May need to change that for this to work, or create a token with broader permissions and pass that around via secrets. I can run some tests on my fork with more limited settings to see what happens.

Reply:
Good catch. Needed to add
```
permissions:
  contents: write
  pull-requests: write
```
to work with the current default permissions. We do also need that "Allow GitHub Actions to create and approve pull requests" setting checked. I'll check if we can flip that here.